core/lib/spree/event/adapters/active_support_notifications.rb

```diff
@@ -29,6 +31,17 @@ def listeners_for(names)
             memo[name] = listeners if listeners.present?
           end
         end
+
+        def name_with_suffix(name, suffix)
+          case name
+          when String
+            name.end_with?(suffix) ? name : [name, suffix].join
+          when Regexp
+            Regexp.new("#{name}#{Regexp.escape(suffix)}$")
```

Comment:
We should let it be known that different behavior can happen based on the type here:

For example:
```ruby
name_with_suffix('order_finalized.spree', 'spree')
#=> 'order_finalized.spree'

# But this differs if regex is used

name_with_suffix(/order_finalized\.spree/, 'spree')
#=> /order_finalized\.spree\.spree$/
```

The method tries to check if spree is included in the string case. if it isn't then adds it.
However in the case of regexp, adds it no matter what.

Reply:
Would it be best to deprecate the explicit inclusion of `suffix` in the event name and always append it in `name_with_suffix`?

Reply:
> We should let it be known that different behavior can happen based on the type

Sure. 
BTW I think the different behavior for String and Regexp event names is not much of a real problem, `String` and `Regexp` are different things, so some different behavior can be tolerated IMO.

>Would it be best to deprecate the explicit inclusion of suffix in the event name and always append it in name_with_suffix?

mmm not sure I understand what you mean here, can you explain with more detail?

Reply:
Sorry for the confusion, what I mean is instead of conditionally appeneding the suffix, we always append the suffix.

```ruby
# Conditional
name_with_suffix('my_event.spree')
#=> 'my_event.spree'
```

Changes to:

```ruby
# Always
name_with_suffic('my_event.spree')
#=> 'my_event.spree.spree'
```

This would allow the same behaviour to occur for both types passed in since it is expected that spree will be appended always.

I'm ok with the current code as is though, so this is just for discussion

Reply:
Thanks for the clarification. I think both approaches have merits, but I am rather undecided about which one is better.  Maybe somebody has more strong opinions on this, but IMO it does not change really much in the end.

Reply:
I also don't like that the behavior is different if you pass different things to this method. I prefer to make it explicit by not calling `name_with_suffix` at all if a regex is passed + changing YARD documentation for the `subscribe` method and the guides' text explaining how to use this feature.

Reply:
See https://github.com/solidusio/solidus/pull/3519 for resolution of this comment.